torch/csrc/jit/source_range.h

```diff
@@ -67,6 +68,17 @@ struct Source {
     return starting_line_no_;
   }
 
+  // Serialize as Tuple[str, Optional[str], int, List[int]]
+  std::shared_ptr<c10::IValue> __getstate__() {
```

Comment:
shared_ptr<IValue>? that is almost always wrong.

Reply:
Also this code should not be a member of SourceRange or defined in source_range.h it adds a huge dependency of ivalue.h/type.h etc. to source range, which previously included almost no headers. This will dramatically slow build times.